boofuzz/sessions.py

```diff
@@ -1000,6 +1079,16 @@ def _pre_send(self, target):
             target (session.target): Target we are sending data to
         """
 
+        try:
+            for monitor in target.monitors:
+                self._fuzz_data_logger.open_test_step("Monitor {}.pre_send()".format(str(monitor)))
+                monitor.pre_send(target=target, fuzz_data_logger=self._fuzz_current_case, session=self)
+        except Exception:
+            self._fuzz_data_logger.log_error(
+                constants.ERR_CALLBACK_FUNC.format(func_name="{}.pre_send()".format(str(monitor)))
+                + traceback.format_exc()
+            )
```

Comment:
Shouldn't the try except be inside the loop over the monitors as we use `monitor` inside the except clause?

PyCharm complains about a possible reference before assignment, and I guess that's a somewhat valid concern here with the very brought exception clause.

Reply:
Yeah, totally. Don't know why I left it like that.